File: Documentation/5.4/Samples/python/Indexes/Storing.py
from ravendb import AbstractIndexCreationTask, PutIndexesOperation
from ravendb.documents.indexes.definitions import FieldStorage, IndexDefinition, IndexFieldOptions

from examples_base import ExampleBase


# region storing_1
class Employees_ByFirstAndLastName(AbstractIndexCreationTask):
    def __init__(self):
        super().__init__()
        self.map = "from employee in docs.Employees select new { employee.FirstName, employee.LastName }"
        self._store("FirstName", FieldStorage.YES)
        self._store("LastName", FieldStorage.YES)


# endregion


class Storing(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_storing(self):
        with self.embedded_server.get_document_store("Storing") as store:
            with store.open_session() as session:
                # region storing_2
                store.maintenance.send(
                    PutIndexesOperation(
                        IndexDefinition(
                            name="Employees_ByFirstAndLastName",
                            maps={
                                """
                    from employee in docs.Employees
                    select new
                    {
                        employee.FirstName,
                        employee.LastName
                    }
                    """
                            },
                            fields={
                                "FirstName": IndexFieldOptions(storage=FieldStorage.YES),
                                "LastName": IndexFieldOptions(storage=FieldStorage.YES),
                            },
                        )
                    )
                )
                # endregion

File: Documentation/5.4/Samples/php/Indexes/Storing.php
<?php

namespace RavenDB\Samples\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\AbstractIndexCreationTask;
use RavenDB\Documents\Indexes\FieldStorage;
use RavenDB\Documents\Indexes\IndexDefinition;
use RavenDB\Documents\Indexes\IndexFieldOptions;
use RavenDB\Documents\Operations\Indexes\PutIndexesOperation;

# region storing_1
class Employees_ByFirstAndLastName extends AbstractIndexCreationTask
{
    public function __construct()
    {
        parent::__construct();

        $this->map =  "docs.Employees.Select(employee => new {" .
            "    FirstName = employee.FirstName," .
            "    LastName = employee.LastName" .
            "})";

        $this->store('FirstName', FieldStorage::yes());
        $this->store('LastName', FieldStorage::yes());
    }
}
# endregion

class Storing
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region storing_2
            $indexDefinition = new IndexDefinition();
            $indexDefinition->setName("Employees_ByFirstAndLastName");
            $indexDefinition->setMaps([
                "docs.Employees.Select(employee => new {" .
                "    FirstName = employee.FirstName," .
                "    LastName = employee.LastName" .
                "})"
            ]);

            $fields = [];

            $firstNameOptions = new IndexFieldOptions();
            $firstNameOptions->setStorage(FieldStorage::yes());
            $fields['FirstName'] = $firstNameOptions;

            $lastNameOptions = new IndexFieldOptions();
            $lastNameOptions->setStorage(FieldStorage::yes());
            $fields['LastName'] = $lastNameOptions;

            $indexDefinition->setFields($fields);

            $store->maintenance()->send(new PutIndexesOperation($indexDefinition));
            # endregion
        } finally {
            $store->close();
        }
    }
}
